### ISSUE TITLE
Something funky with indexes and DataFrames

### ISSUE THREAD

**Issue by CamDavidsonPilon** [matthewwardrop/formulaic#31]
Hi @matthewwardrop! 
Something came up in [lifelines](https://github.com/CamDavidsonPilon/lifelines/issues/1215) related to Dataframe indexes and `transform`. Here's a repo example:

```
from formulaic import Formula

design_info = Formula("1")

df = pd.DataFrame(np.arange(5), index=[0, 2, 4, 6, 8])

print(design_info.get_model_matrix(df)) # should have nulls in half of it
```

**Comment by matthewwardrop** [comment]
Thanks for reporting @CamDavidsonPilon . Will take a look at fixing it now!

**Comment by matthewwardrop** [comment]
Whoops. Some series were being generated with the default index (and being conflated with the original index). This was fixed in 6567f44217a5e3089866b193b9a865e2f3a4de96, and shipped as part of version 0.2.2 .

Let me know if anything else crops up!

**Comment by CamDavidsonPilon** [comment]
Thanks for the fast response!
